diff --git a/node_modules/@giphy/react-native-sdk/giphy-react-native-sdk.podspec b/node_modules/@giphy/react-native-sdk/giphy-react-native-sdk.podspec
index 41cf883..0118897 100644
--- a/node_modules/@giphy/react-native-sdk/giphy-react-native-sdk.podspec
+++ b/node_modules/@giphy/react-native-sdk/giphy-react-native-sdk.podspec
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
 
   s.source_files = "ios/**/*.{h,m,mm,swift}"
 
-  s.dependency "Giphy", "2.2.9"
+  s.dependency "Giphy", "2.2.11"
 
   s.pod_target_xcconfig = {
     "DEFINES_MODULE" => "YES"
