diff --git a/node_modules/react-native-immersive-mode/android/build.gradle b/node_modules/react-native-immersive-mode/android/build.gradle
index e642408..93bcc2c 100644
--- a/node_modules/react-native-immersive-mode/android/build.gradle
+++ b/node_modules/react-native-immersive-mode/android/build.gradle
@@ -23,6 +23,8 @@ android {
     lintOptions {
         abortOnError false
     }
+
+    namespace "com.rnimmersivemode"
 }
 
 repositories {
