diff --git a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
--- a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
+++ b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
@@ -53,7 +53,7 @@ static UIInterfaceOrientationMask _orientationMask = UIInterfaceOrientationMaskA
 - (void)dealloc
 {
     [[NSNotificationCenter defaultCenter] removeObserver:self];
-    [self removeListeners:1];
+    // Upstream PR: https://github.com/wonday/react-native-orientation-locker/pull/316
 }

 - (UIInterfaceOrientation)getInterfaceOrientation
