diff --git a/node_modules/react-native-immersive/index.js b/node_modules/react-native-immersive/index.js
index 55dab57..110260b 100644
--- a/node_modules/react-native-immersive/index.js
+++ b/node_modules/react-native-immersive/index.js
@@ -18,7 +18,13 @@ const Immersive = Platform.OS === 'android' ? {
     isListenerEnabled = true
     RNImmersive.addImmersiveListener()
   },
-  removeImmersiveListener: (listener) => DeviceEventEmitter.removeListener('@@IMMERSIVE_STATE_CHANGED', listener)
+    removeImmersiveListener: (listener) => {
+        const immersiveListener = DeviceEventEmitter.addListener('@@IMMERSIVE_STATE_CHANGED', listener);
+
+        return () => {
+            immersiveListener.remove();
+        }
+    }
 } : {
   on: unSupportedError,
   off: unSupportedError,
