--- /usr/lib/prosody/modules/mod_bosh.lua	2015-12-16 14:28:34.000000000 -0600
+++ /usr/lib/prosody/modules/mod_bosh.lua	2015-12-22 10:45:59.818197967 -0600
@@ -294,6 +294,9 @@
 
 		session.log("debug", "BOSH session created for request from %s", session.ip);
 		log("info", "New BOSH session, assigned it sid '%s'", sid);
+		
+		hosts[session.host].events.fire_event(
+			"bosh-session", { session = session, request = request });
 
 		-- Send creation response
 		local creating_session = true;
